src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,51 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index !== -1) {
+      Animal.alive.splice(index, 1);
+    }
+  }
```

Comment:
[CODE PERFORMANCE] The `die` method uses `indexOf` and `splice` to remove the animal from the `Animal.alive` array. This is not optimal as it involves searching through the array and then modifying it, which is inefficient, especially for large arrays. A better approach would be to use the `filter` method to remove dead animals from the `Animal.alive` array. This can be done directly in the `checkHealth` method or in a separate method that is called when necessary.